tests/wallet/did_wallet/test_did.py

```diff
@@ -793,10 +793,8 @@ async def test_did_transfer(self, self_hostname, two_wallet_nodes, with_recovery
         assert metadata["GitHub"] == "测试"
 
         # Test match_hinted_coin
-        assert await did_wallet_2.match_hinted_coin(
-            await did_wallet_2.get_coin(),
-            new_puzhash,
-        )
+        if with_recovery:
+            assert await did_wallet_2.match_hinted_coin(await did_wallet_2.get_coin(), new_puzhash)
```

Comment:
Why this change?

Reply:
In the without recovery info mode, the recovery setting will be reset so the match will fail.

Reply:
But why did it break in this PR?  This is supposed to be an optimization right so why did the behavior change?

Reply:
I thought there is a bug, revert the change.